webdriver/tests/document_handling/test_execute_script.py

```diff
@@ -0,0 +1,189 @@
+
+
+def test_return_boolean(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert True == session.execute_script("return true")
+
+
+def test_return_string(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert "foo" == session.execute_script("return 'foo'")
+
+
+def test_return_array(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert [1, 2] ==  session.execute_script("return [1, 2]")
+    assert [1.25, 1.75] ==  session.execute_script("return [1.25, 1.75]")
+    assert [True, False] == session.execute_script("return [true, false]")
+    assert ["foo", "bar"] == session.execute_script("return ['foo', 'bar']")
+    assert [1, 1.5, True, "foo"] == session.execute_script("return [1, 1.5, true, 'foo']")
+    assert [1, [2]] == session.execute_script("return [1, [2]]")
+
+
+def test_return_object(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert {"foo": 1} == session.execute_script("return {foo: 1}")
+    assert {"foo": 1.5} == session.execute_script("return {foo: 1.5}")
+    assert {"foo": True} == session.execute_script("return {foo: true}")
+    assert {"foo": "bar"} == session.execute_script("return {foo: 'bar'}")
+    assert {"foo": [1, 2]} == session.execute_script("return {foo: [1, 2]}")
+    assert {"foo": {"bar": [1, 2]}} == \
+        session.execute_script("return {foo: {bar: [1, 2]}}")
+
+
+def test_no_return_value(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert None == session.execute_script("true")
+
+
+def test_argument_null(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert None == session.execute_script("return arguments[0]", args=[None])
+
+
+def test_argument_number(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert 1 == session.execute_script("return arguments[0]", args=[1])
+    assert 1.5 == session.execute_script("return arguments[0]", args=[1.5])
+
+
+def test_argument_boolean(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert True == session.execute_script("return arguments[0]", args=[True])
+
+
+def test_argument_string(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert "foo" == session.execute_script("return arguments[0]", args=["foo"])
+
+
+def test_argument_array(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert [1, 2] == session.execute_script("return arguments[0]", args=[[1, 2]])
+
+
+def test_argument_object(session):
+    session.url = inline("""<title>JS testing</title>""")
+    assert {"foo": 1} == session.execute_script(
+            "return arguments[0]", args=[{"foo": 1}])
+
+globals = set([
+              "document",
+              "navigator",
+              "window",
+              ])
+
+def test_access_todefault_globals(session):
+    session.url = inline("""<title>JS testing</title>""")
+    for property in globals:
```

Comment:
This can be turned into a pytst [parameterised](https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize) test.